### PR TITLE
Iotbzh/for upstream/can

### PR DESCRIPTION
**Pull request by julien-massot**
dry run

### CODE REVIEW

File: drivers/can/can_rcar.c [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+
+static inline int can_rcar_attach(const struct device *dev,
+				  can_rx_callback_t cb,
+				  void *cb_arg,
+				  const struct zcan_filter *filter)
+{
+	struct can_rcar_data *data = DEV_CAN_DATA(dev);
+	int i;
+
+	for (i = 0; i < CONFIG_CAN_RCAR_MAX_FILTER; i++)
```
Comment (julien-massot) [inline drivers/can/can_rcar.c]: missing brace

File: drivers/can/can_rcar.c [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
+	data->state_change_isr = isr;
+}
+
+static enum can_state can_rcar_get_state(const struct device *dev,
+					 struct can_bus_err_cnt *err_cnt)
+{
+	const struct can_rcar_cfg *config = DEV_CAN_CFG(dev);
+	struct can_rcar_data *data = DEV_CAN_DATA(dev);
+
+	if (err_cnt) {
```
Comment (julien-massot) [inline drivers/can/can_rcar.c]: != NULL

File: drivers/can/can_rcar.c [hunk 3]
```diff
@@ -0,0 +1,1042 @@
...
+#define RCAR_CAN_EIER             0x084C
+
+/* Interrupt Enable Register */
+#define RCAR_CAN_IER              0x0860
+#define RCAR_CAN_IER_ERSIE        BIT(5)      /* Error Interrupt Enable Bit */
+#define RCAR_CAN_IER_RXFIE        BIT(4)      /* Rx FIFO Interrupt Enable Bit */
+#define RCAR_CAN_IER_TXFIE        BIT(3)      /* Tx FIFO Interrupt Enable Bit */
+
+/* Interrupt Status Register */
+#define RCAR_CAN_ISR		  0x0861
```
Comment (aaillet) [inline drivers/can/can_rcar.c]: indent ?

File: drivers/can/can_rcar.c [hunk 4]
```diff
@@ -0,0 +1,1042 @@
...
+	case CAN_SILENT_MODE:
+		tcr = RCAR_CAN_TCR_LISTEN_ONLY | RCAR_CAN_TCR_TSTE;
+		break;
+	/*Controller is in loopback mode (receive own messages)*/
+	case CAN_LOOPBACK_MODE:
+		tcr = RCAR_CAN_TCR_INT_LOOP | RCAR_CAN_TCR_TSTE;
+		break;
+	/*Combination of loopback and silent*/
+	case CAN_SILENT_LOOPBACK_MODE:
+		ret = -ENOSYS;
```
Comment (julien-massot) [inline drivers/can/can_rcar.c]: -ENOTSUP

File: drivers/can/can_rcar.c [hunk 5]
```diff
@@ -0,0 +1,1042 @@
...
+
+/* Receive FIFO Pointer Control Register */
+#define RCAR_CAN_RFPCR            0x0849
+
+/* Transmit FIFO Control Register */
+#define RCAR_CAN_TFCR             0x084A
+#define RCAR_CAN_TFCR_TFE         BIT(0)      /* Transmit FIFO Enable */
+#define RCAR_CAN_TFCR_TFUST       (7 << 1)    /* Transmit FIFO Unsent Message */
+					      /* Number Status Bits */
+#define RCAR_CAN_TFCR_TFUST_SHIFT 1	      /* Offset of Tx FIFO Unsent */
```
Comment (julien-massot) [inline drivers/can/can_rcar.c:113]: indent

File: drivers/can/can_rcar.c (hunk 3 above)
Comment (julien-massot) [inline drivers/can/can_rcar.c]: indent

File: drivers/can/can_rcar.c [hunk 6]
```diff
@@ -0,0 +1,1042 @@
...
+			continue; /* filter did not match */
+		}
+		/* Make a temporary copy in case the user
+		 * modifies the message.
+		 */
+		tmp_msg = *msg;
+		data->rx_callback[i](&tmp_msg, data->rx_callback_arg[i]);
+	}
+	k_mutex_unlock(&data->rx_mutex);
+};
```
Comment (aaillet) [inline drivers/can/can_rcar.c]: other functions are only closed by "}" not "};", please choose one or the other

File: drivers/can/can_rcar.c [hunk 7]
```diff
@@ -0,0 +1,1042 @@
...
+			unsent = (unsent & RCAR_CAN_TFCR_TFUST) >>
+				RCAR_CAN_TFCR_TFUST_SHIFT;
+			if (data->tx_head - data->tx_tail <= unsent) {
+				break;
+			}
+			can_rcar_tx_done(dev);
+		}
+		k_mutex_unlock(&data->inst_mutex);
+	}
+};
```
Comment (aaillet) [inline drivers/can/can_rcar.c]: "};" or "}"

File: drivers/can/can_rcar.c [hunk 8]
```diff
@@ -0,0 +1,1042 @@
...
+		break;
+	/*Combination of loopback and silent*/
+	case CAN_SILENT_LOOPBACK_MODE:
+		ret = -ENOSYS;
+		goto unlock;
+	}
+
+	/* Writing to TCR registers must be done in halt mode */
+	ret = can_rcar_enter_halt_mode(config);
+	if (ret) {
```
Comment (aaillet) [inline drivers/can/can_rcar.c:610]: if (ret != 0)

File: drivers/can/can_rcar.c (hunk 2 above)
Comment (aaillet) [inline drivers/can/can_rcar.c]: bool cast ? is it authorized on the type ?

File: drivers/can/can_rcar.c [hunk 9]
```diff
@@ -0,0 +1,1042 @@
...
+	data->tx_tail = 0;
+	data->int_thread_stack = can_rcar_thread_stack;
+	memset(data->rx_callback, 0, sizeof(data->rx_callback));
+	data->state = CAN_ERROR_ACTIVE;
+	data->state_change_isr = NULL;
+
+	/* reset the registers */
+	ret = clock_control_off(config->clock_dev,
+				(clock_control_subsys_t *)&config->mod_clk);
+	if (ret < 0) {
```
Comment (aaillet) [inline drivers/can/can_rcar.c:850]: if (ret != 0) for homogeneity ?

Comment (aaillet) [reply drivers/can/can_rcar.c:850]: l854 + 860 + 866 + 872 + 892 + 898 + 903 + 952

File: drivers/can/can_rcar.c (hunk 6 above)
Comment (julien-massot) [reply drivers/can/can_rcar.c]: fixed thanks

File: drivers/can/can_rcar.c (hunk 7 above)
Comment (julien-massot) [reply drivers/can/can_rcar.c]: fixed

File: drivers/can/can_rcar.c (hunk 2 above)
Comment (julien-massot) [reply drivers/can/can_rcar.c]: fixed

File: drivers/can/can_rcar.c (hunk 1 above)
Comment (julien-massot) [reply drivers/can/can_rcar.c]: fixed

File: drivers/can/can_rcar.c (hunk 9 above)
Comment (julien-massot) [reply drivers/can/can_rcar.c:850]: return 0 on success, negative errno on failure
sounds good

**Comment by aaillet** [comment]
There is a lack of documentation update for this PR

**Pull request closed without merging**